integration-cli/docker_api_swarm_test.go

```diff
@@ -471,6 +471,67 @@ func (s *DockerSwarmSuite) TestApiSwarmLeaderProxy(c *check.C) {
 	}
 }
 
+func (s *DockerSwarmSuite) TestApiSwarmLeaderElection(c *check.C) {
+	// Create 3 nodes
+	d1 := s.AddDaemon(c, true, true)
+	d2 := s.AddDaemon(c, true, true)
+	d3 := s.AddDaemon(c, true, true)
+
+	// assert that the first node we made is the leader, and the other two are followers
+	c.Assert(d1.getNode(c, d1.NodeID).ManagerStatus.Leader, checker.True)
+	c.Assert(d1.getNode(c, d2.NodeID).ManagerStatus.Leader, checker.False)
+	c.Assert(d1.getNode(c, d3.NodeID).ManagerStatus.Leader, checker.False)
+
+	leader := d1
+
+	// stop the leader
+	leader.Stop()
+
+	// wait for an election to occur
+	var newleader *SwarmDaemon
+
+	timeout := time.After(defaultReconciliationTimeout)
+
+	// loop until leader
+	for newleader == nil {
```

Comment:
Did this fail without the for loop? `d2.getNode` should not return if there is no leader. The timeout handling isn't quite perfect atm but in this case, the timeout would fail the test anyway as `getNode` would cause an error.